docs/proposals/going-towards-to-crd.md

```diff
@@ -0,0 +1,104 @@
+  | `PrepareForUpdate` methods and defaults schemas via `defaulter-gen` | Unified and realised only via **MutatingWebhookConfiguration** |
+
+- defining services in Helm Chart - different RBACs, secrets, deployments, services, etc.
+
+- underlying Kubernetes different version constraint
+
+> **NOTE:** As you can see above the Aggregated API use the validation and mutation in a different way. Sometimes same logic is split across different pkgs. In CRDs we unified that and copied it directly to the webhook domain. If we want to support both the API and CRD concept then we need to invest some time to extract it to some common libs. Then in both places use it with an overhead of adjusting the generic interfaces to a custom one. Additionally, after removing the API Server, we need to migrate them back to the domain as having it extracted will only mess the code and additional abstraction layer can confuse other developers.  
+ 
+> **NOTE:** The described differences are those that we see from the short walk-through. Finally, there may be more differences - especially if we will support adding features both for api-server and CRDs.
+ 
+**From the technical point of view having Aggregated API and CRDs in the same code-base it's not a feature gate only in the `service-catalog` binary but in the whole eco-system.** 
+
+## Alternative Solution
+
+Before merging the CRDs solution into the master, create a branch `release-0.1` with the last release of Service Catalog with Aggregated API server. 
+On the `release-0.1` branch we are still doing the **bug fixing** for this version of Service Catalog but features are not introduced. When some bug will be found then we can fix it and still in an easy way create a release with version `0.1.x` 
+
+In the master branch, we have Service Catalog with the CRD approach. New releases are created with the `0.2.x` version. On master branch, we are doing both the bug fixing and adding new features.
+
+Such strategy will help us to get new customers cause we will show that the CRD is our new direction on which we want to focus. 
+ 
+On the other hand, existing Service Catalog customers will see that the old solution is deprecated and the new features will be available only in the new approach. Thanks to that they will consider updating their system to the newest version as soon as possible cause the goal is to use the CRD solution as Kubernetes community recommends that.
+
+We can set a reasonable time for supporting bug fixing for the Service Catalog with Aggregated API, e.g. 1 year.
+
+With such solution, problems described in the previous section are mitigated.  
+     
+## Migration Support
+
+If we want to convince existing customers to use the new Service Catalog version then we need to create a migration guide/scripts. The migration needs to be as simple as possible.
+
+The migration will be simpler when we will hardcode for a given functionality in Aggregated API. 
```

Comment:
hmm, maybe just say that Api-server solution will have just bug fixes and migration to the CRDs will be then simpler ?